lib/ae_page_objects/core/dsl.rb

```diff
@@ -20,7 +20,9 @@ def element(name, options = {}, &block)
 
       self.element_attributes[name.to_sym] = klass
 
-      define_method name do |&block|
+      define_method name do |&block_for_define_method|
```

Comment:
@dtognazzini 
I think raising an exception when passing a block is more conservative, that help developer to find the wrong usage of a method.

For example, an exception will be thrown if developer write code like this:

```ruby
element :some_element do
  element :nested
end

some_element do { |e| e.text == 'xxx' }
```

Since in Ruby you can pass a block to any method, that doesn't make sense to raise an exception for every method that does not do anything with the block. 

Maybe we can only do the raise exception thing for the methods that may let developer think it can accept block but not? 
like `visit` method(https://github.com/appfolio/ae_page_objects/blob/master/lib/ae_page_objects/concerns/visitable.rb#L25)?